### PR TITLE
#11: Converted DownloadUDF to SET UDF

### LINKED ISSUE #11: Convert DownloaderUDF to SET UDF
- the models to be downloaded can be given in the table below.
- this table might include _bucketfs_connection_name_ and _model_name_ as columns
- to scale downloader udf like that, we need to convert it to SET UDF

### PR DESCRIPTION
**Pull request by umitbuyuksahin**
Fixes #11

### CODE REVIEW

File: tests/integration_tests/with_db/udfs/test_model_downloader_udf_script.py [hunk 1]
```diff
@@ -3,32 +3,50 @@
...
+        ))
+
     bucketfs_files = []
     try:
+        query = f"SELECT TE_MODEL_DOWNLOADER_UDF(" \
+                f"t.model_name, " \
+                f"t.sub_dir, " \
+                f"t.bucketfs_conn_name" \
+                f") FROM (VALUES {str(tuple(input_data))} " \
+                f"AS t(model_name, sub_dir, bucketfs_conn_name));"
```
Comment (Nicoretti) [inline tests/integration_tests/with_db/udfs/test_model_downloader_udf_script.py:34]:
```suggestion
        query = (
                f"SELECT TE_MODEL_DOWNLOADER_UDF(" 
                f"t.model_name, "
                f"t.sub_dir, "
                f"t.bucketfs_conn_name"
                f") FROM (VALUES {str(tuple(input_data))} "
                f"AS t(model_name, sub_dir, bucketfs_conn_name));"
        )
```

File: tests/unit_tests/udfs/test_model_downloader_udf.py [hunk 2]
```diff
@@ -83,25 +82,37 @@ def test_model_downloader():
...
+                   == MODEL_FILE_DATA_MAP["model_file1.txt"] \
+                   and bucketfs_location_read.read_file_from_bucketfs_to_string(
+                str(relative_model_path.joinpath("model_file2.txt"))) \
+                   == MODEL_FILE_DATA_MAP["model_file2.txt"] \
+                   and bucketfs_location_read.read_file_from_bucketfs_to_string(
+                str(relative_model_path.joinpath("tokenizer_file1.txt"))) \
+                   == TOKENIZER_FILE_DATA_MAP["tokenizer_file1.txt"] \
+                   and bucketfs_location_read.read_file_from_bucketfs_to_string(
+                str(relative_model_path.joinpath("tokenizer_file2.txt"))) \
+                   == TOKENIZER_FILE_DATA_MAP["tokenizer_file2.txt"]
```
Comment (Nicoretti) [inline tests/unit_tests/udfs/test_model_downloader_udf.py:118]: Even though that I am  fan of having just a single assert, I think in this case multiple asserts will be easier to understand and investigate. I personally think multiple simple asserts are better than a single very complex one.

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -178,8 +178,8 @@ There are two ways to install the language container: (1) using a python script
 ```buildoutcfg
 python -m exasol_transformers_extension.deploy scripts
     --dsn <DB_HOST:DB_PORT> \
-    --db-user <DB_USER> \
-    --db-pass <DB_PASSWORD> \
+    --user <DB_USER> \
+    --pass <DB_PASSWORD> \
```
[the remark this answers is not in the document]
Comment (umitbuyuksahin) [reply doc/user_guide/user_guide.md:182]: thanks :+1: , that might be handled in another ticket. It is used in this way in other places as well.

File: tests/unit_tests/udfs/test_model_downloader_udf.py (hunk 2 above)
Comment (umitbuyuksahin) [reply tests/unit_tests/udfs/test_model_downloader_udf.py:118]: I'm in favor of it being this way, it is much easier to follow the errors.

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**